sdk/python/foundation-models/system/content-safety/score/score.py

```diff
@@ -0,0 +1,407 @@
+    elif isinstance(sample_output, dict):
+        param_arg = {}
+        for key, value in sample_output.items():
+            param_arg[key] = NumpyParameterType(value, enforce_shape=False)
+        output_param = StandardPythonParameterType(param_arg)
+    else:
+        output_param = PandasParameterType(
+            sample_output, enforce_shape=False, orient="records"
+        )
+
+# we use mlflow.pyfunc's load_model function because it has a predict function on it we need for inferencing
+model = load_model(model_path)
+
+
+def init():
+    global aacs_client
+    endpoint = os.environ.get("CONTENT_SAFETY_ENDPOINT")
+    key = os.environ.get("CONTENT_SAFETY_KEY")
+
+    # Create an Content Safety client
+    aacs_client = ContentSafetyClient(endpoint, AzureKeyCredential(key))
+
+    _logger.info("init")
+
+
+async def async_analyze_text_task(client, request):
+    loop = asyncio.get_event_loop()
+    executor = async_rate_limiter.get_executor()
+    sem = async_rate_limiter.get_semaphore()
+    await sem.acquire()
+    async with async_rate_limiter.get_limiter():
+        response = await loop.run_in_executor(executor, client.analyze_text, request)
+        sem.release()
+        severity = analyze_response(response)
+        return severity
+
+
+def analyze_response(response):
+    severity = 0
+
+    if response.hate_result is not None:
+        _logger.info("Hate severity: {}".format(response.hate_result.severity))
+        severity = max(severity, response.hate_result.severity)
+    if response.self_harm_result is not None:
+        _logger.info("SelfHarm severity: {}".format(response.self_harm_result.severity))
+        severity = max(severity, response.self_harm_result.severity)
+    if response.sexual_result is not None:
+        _logger.info("Sexual severity: {}".format(response.sexual_result.severity))
+        severity = max(severity, response.sexual_result.severity)
+    if response.violence_result is not None:
+        _logger.info("Violence severity: {}".format(response.violence_result.severity))
+        severity = max(severity, response.violence_result.severity)
+
+    return severity
+
+
+def analyze_text(text):
+    # Chunk text
+    chunking_utils = CsChunkingUtils(chunking_n=1000, delimiter=".")
+    split_text = chunking_utils.split_by(text)
+
+    tasks = []
+    for i in split_text:
+        request = AnalyzeTextOptions(text=i)
+        tasks.append(async_analyze_text_task(aacs_client, request))
+
+    done, pending = asyncio.get_event_loop().run_until_complete(
+        asyncio.wait(tasks, timeout=60)
+    )
+
+    if len(pending) > 0:
+        # not all task finished, assume failed
+        return 2
+
+    return max([d.result() for d in done])
+
+
+def get_safe_response(result):
+    jsonable_result = _get_jsonable_obj(result, pandas_orient="records")
+    return_result = []
+    for d in jsonable_result:
+        result = d[0]
+        if analyze_text(result) > 2:
+            return_result.append({0: ""})
```

Comment:
instead of NULL, would not it be better to put something like "content blocked", that will make testing little bit simpler?

Reply:
Keep "" then talk to PM